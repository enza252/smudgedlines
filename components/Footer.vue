<template>
  <v-footer
    app
    class="pb-0"
  >
    <v-container style="width: 100%">
      <v-row>
        <v-col cols="5" class="pl-2 pt-0 pb-0" style="height: 26px !important;">
          <template v-for="(social, index) in socials">
            <v-tooltip top :key="index">
              <template v-slot:activator="{ on, attrs }">
                <a :href="social.url">
                  <v-icon  v-if="social.icon" v-on="on" v-bind="attrs" style="margin-bottom: 12px !important;">
                    {{social.icon}}
                  </v-icon>

                  <v-img v-else :src="social.img" height="22" width="22" v-on="on" v-bind="attrs" :class="social.classes"/>

                </a>
              </template>
              Visit our {{social.tooltipText}} Page
            </v-tooltip>

          </template>

        </v-col>
        <v-spacer/>
        <v-col :cols="mobile ? 7 : 6" style="text-align: end" class="pa-0">
          <span v-if="!mobile">&copy; {{ new Date().getFullYear() }} made with <v-icon>mdi-heart</v-icon> by Smudged Lines</span>
          <span v-else>&copy; {{ new Date().getFullYear() }} Smudged Lines</span>
        </v-col>
      </v-row>
    </v-container>
  </v-footer>
</template>
<script>
import { defineComponent } from '@nuxtjs/composition-api'
import { mdiInstagram } from '@mdi/js'
import depopLogo from '/static/socials/depop-logo-transparent-24x24.svg'

export default defineComponent({
  name: 'Footer',
  components: {
    depopLogo
  },
  data: () => ({
    socials: [{
      icon: mdiInstagram,
      tooltipText: 'Instagram',
      url: 'https://www.instagram.com/smudgedlines/',
      classes: 'pr-2'
    },
    {
      img: '/socials/depop-logo-transparent.png',
      tooltipText: 'Depop',
      url: 'https://www.depop.com/smudgedlines/',
      classes: 'd-inline-flex flex-grow-1 align-end align-self-end align-content-end pl-2'
    }
    ]
  }),
  computed: {
    mobile () {
      return this.$vuetify.breakpoint.smAndDown
    }
  }
})
</script>

<style lang="sass">

</style>
