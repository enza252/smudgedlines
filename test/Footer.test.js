import { render, screen, waitFor } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'
import Footer from '@/components/Footer'
import Vuetify from 'vuetify'

const options = {
  routes: [],
  vuetify: new Vuetify()
}

describe('tests the footer renders correctly on large screens', () => {
  it('Renders the Header and checks against a snapshot', async () => {
    const { asFragment } = await render(Footer, options)
    expect(asFragment).toMatchSnapshot()
  })
  it('tests the text and icons are present within the component', async () => {
    await render(Footer, options)
    expect(screen.getByTestId('footer-text-right').textContent).toContain(`© ${new Date().getFullYear()} made with  by Smudged Lines`)
    expect(screen.getByTestId('depop')).toBeInTheDocument()
    expect(screen.getByTestId('instagram')).toBeInTheDocument()
    await userEvent.hover(screen.getByTestId('depop'))
    await waitFor(() => {
      expect(screen.getByText('Visit our Depop Page')).toBeVisible()
    })
    await userEvent.hover(screen.getByTestId('instagram'))
    await waitFor(() => {
      expect(screen.getByText('Visit our Instagram Page')).toBeVisible()
    })
  })
})

describe('tests the footer renders correctly on small screens', () => {
  it('renders the footer and checks the appropriate tagline is rendered', async () => {
    Object.defineProperty(window, 'innerHeight', {
      writable: true,
      configurable: true,
      value: 150
    })
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: 768
    })
    window.dispatchEvent(new Event('resize'))
    expect(window.innerHeight).toBe(150)
    expect(window.innerWidth).toBe(768)
    await render(Footer, options)
    expect(screen.getByTestId('footer-text-right-mobile').textContent).toContain(`© ${new Date().getFullYear()} Smudged Lines`)
  })
})
