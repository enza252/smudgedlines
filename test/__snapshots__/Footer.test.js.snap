// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the footer renders correctly on large screens Renders the Header and checks against a snapshot 1`] = `undefined`;
